avocado/utils/disk.py

```diff
@@ -12,15 +12,40 @@
 # This code was inspired in the autotest project,
 #
 # client/base_utils.py
+#
+# Copyright: 2018 IBM
+# Authors : Praveen K Pandey <praveen@linux.vnet.ibm.com>
 
 
 """
 Disk utilities
 """
 
+
 import os
+import json
+from . import process
 
 
 def freespace(path):
     fs_stats = os.statvfs(path)
     return fs_stats.f_bsize * fs_stats.f_bavail
+
+
+def get_disk_blocksize(path):
+    """Return the disk block size, in bytes"""
+    fs_stats = os.statvfs(path)
+    return fs_stats.f_bsize
+
+
+def get_disks():
+    json_result = process.run('lsblk --json')
+    json_data = json.loads(json_result.stdout_text)
+    disk_list = []
+    iterator = 0
+    while iterator < len(json_data['blockdevices']):
+        disk = '/dev/' + json_data['blockdevices'][iterator]['name']
+        disk_list.append(str(disk))
+        iterator += 1
+
+    return disk_list
```

Comment:
Yep, using machine readable output is way better. Would you please consider using list comprehension rather than iterating through result?

```diff
diff --git a/avocado/utils/disk.py b/avocado/utils/disk.py
index 8e6fa617..2281a640 100644
--- a/avocado/utils/disk.py
+++ b/avocado/utils/disk.py
@@ -41,11 +41,4 @@ def get_disk_blocksize(path):
 def get_disks():
     json_result = process.run('lsblk --json')
     json_data = json.loads(json_result.stdout_text)
-    disk_list = []
-    iterator = 0
-    while iterator < len(json_data['blockdevices']):
-        disk = '/dev/' + json_data['blockdevices'][iterator]['name']
-        disk_list.append(str(disk))
-        iterator += 1
-
-    return disk_list
+    return ['/dev/%s' % disk['name'] for disk in json_data['blockdevices']]
```

Reply:
@ldoktor thanks a lot for review , it seems good idea will do change and push the changes